0000-qiskit-pulse-schedule-relative-timing.md

```diff
@@ -52,49 +52,107 @@ Factors to consider:
 - Maintenance
 - Compatibility -->
 
-### Notation:
-I am going to use this notation here, where `self` is the `Schedule` initiating the operation in question:
+### Notation
+Here we will use the following notation, where `self` is the `Schedule` initiating the operation in question:
 
  - |C| : the total number of channels in `self`
- - |N<sub>self</sub>| : the total number of instructions in `self`
- - |N<sub>new</sub>| : the total number of instructions in another `Schedule` being added in some way to `self`
- - max(|N<sub>C</sub>|) : the maximum number of instructions on any single channel within `self`
+ - N<sub>self</sub> : the total number of instructions in `self`
+ - N<sub>new</sub> : the total number of instructions in another `Schedule` being added in some way to `self`
+ - N<sub>C</sub> : the number of instructions on channel C within `self`
 
 ### New Schedule methods
 
 | Method        | Runtime        |  Description     |
 |---------------|----------------|------------------|
-|`barrier(self)`| O(\|C\|) | Add `Delay`s across `channels` so that all of the `channels` are occupied until `self.duration(channels)`. |
-|`left_align(self)` | O(\|C\|) | This is the default behavior. We could have it return `self`, or we could remove any pre-pended `Delay`s. The latter is probably better, because `schedule.right_align().left_align()` should yield the original `schedule`. Should we also append `Delay`s? Likely. *Needs more thought* |
-|`center_align(self)` | O(\|C\|) | Pre- and ap-pend equally sized `Delay`s (how to split odd durations?) on each channel so that all channels are occupied until `self.duration()` |
-|`right_align(self)` | O(\|C\|) | Prepend `Delay`s on each channel so that all channels are occupied until `self.duration()`. Should we also remove trailing `Delay`s? |
-
+|`barrier(self, channels=None)`| O(\|C\|) | Add `Delay`s across `channels` so that all of the `channels` are occupied until `self.duration(channels)`. By default, barrier on all channels. |
```

Comment:
I believe this should be implemented as a series of `transforms` on schedules. For example, see my WIP [builder branch](https://github.com/taalexander/qiskit-terra/blob/pulse-builder-interface/qiskit/pulse/transforms.py). We do not want to build too many manipulations directly into a Schedule as this is not extensible. We should only build in what is necessary to allow these tools to be built. 

The builder interface will handle unifying user-level programming and the Schedule IR. I think the Schedule IR should concern itself with being a data structure with timing information and *some* helper methods for common pass operations.

Reply:
ok, I'll remove these both because of this comment and as per discussion today